ic-os/components/setupos-scripts/config.service

```diff
@@ -6,7 +6,7 @@ Before=setupos.service
 [Service]
 Type=oneshot
 RemainAfterExit=true
-ExecStart=/opt/ic/bin/output-wrapper.sh /dev/ttyS0 /opt/ic/bin/config.sh
+ExecStart=/opt/ic/bin/output-wrapper.sh /dev/ttyS0 /opt/ic/bin/config create-setupos-config
 StandardOutput=tty
 StandardError=tty
 
```

Comment:
We want to halt the setupOS installation if config.service fails, correct? 

Should we add
```
Requires=generate-network-config.service setupos.service
After=generate-network-config.service setupos.service
```

Reply:
If I remember there's a weird dependency arrangement here. We want to do config early to get networking up as soon as possible, but we only want to run SetupOS once we've reached login, so that we can drop to a shell when needed.

I think we should keep config early and be sure not to block booting, but maybe there's more we can add to SetupOS to report any failures that have occurred.

Reply:
WDYT of adding a check-config script to setupos.sh that checks for the existence of config.json, logs it if found, and fails otherwise?

Reply:
Updated! see check-config.sh